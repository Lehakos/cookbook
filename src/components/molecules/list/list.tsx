import React from 'react';

import { ListItem } from 'components/atoms';

import * as s from './list.styles';

interface Props {
  items: string[];
  oneLine?: boolean;
  /**
   * The maximum number of items for display
   */
  maxItems?: number;
}

export const List: React.FunctionComponent<Props> = function({
  items,
  oneLine,
  maxItems,
}) {
  const itemsForRender = items.slice(0, maxItems);
  const notShowedItems = items.length - itemsForRender.length;

  return (
    <s.Wrapper>
      {itemsForRender.map(item => (
        <ListItem oneLine={oneLine} key={item}>
          {item}
        </ListItem>
      ))}
      {notShowedItems > 0 && (
        <ListItem oneLine={oneLine}>More {notShowedItems}</ListItem>
      )}
    </s.Wrapper>
  );
};
