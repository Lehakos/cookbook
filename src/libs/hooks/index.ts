export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
