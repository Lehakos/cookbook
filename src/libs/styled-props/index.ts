export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
