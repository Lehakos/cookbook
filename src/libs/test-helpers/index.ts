export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
