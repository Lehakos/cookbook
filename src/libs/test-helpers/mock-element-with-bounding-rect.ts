export const mockElementWithBoundingRect = ({
  width = 100,
  height = 100,
  top = 0,
  left = 0,
}) => {
  const right = left + width;
  const bottom = top + height;

  Object.defineProperties(Element.prototype, {
    offsetLeft: {
      get() {
        return left;
      },
      configurable: true,
    },
    offsetTop: {
      get() {
        return top;
      },
      configurable: true,
    },
    offsetHeight: {
      get() {
        return height;
      },
      configurable: true,
    },
    offsetWidth: {
      get() {
        return width;
      },
      configurable: true,
    },
  });

  Element.prototype.getBoundingClientRect = jest.fn(() => {
    return {
      width,
      height,
      top,
      left,
      bottom,
      right,
    };
  });

  const element = document.createElement('div');

  return element;
};
