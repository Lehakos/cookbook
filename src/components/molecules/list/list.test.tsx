import React from 'react';

import { render } from 'libs/test-helpers';

import { List } from './list';

describe('<List />', () => {
  it('renders properly', () => {
    const { container } = render(<List items={['one', 'two', 'three']} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
