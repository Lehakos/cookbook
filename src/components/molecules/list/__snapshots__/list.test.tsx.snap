// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders properly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1:not(:last-child) {
  margin-bottom: 4px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

.c2::before {
  content: 'A';
  width: 0;
  visibility: hidden;
}

.c2::after {
  height: 4px;
  width: 4px;
  content: '';
  border-radius: 50%;
  background: #F47983;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

<ul
  class="c0"
>
  <li
    class="c1"
  >
    <span
      class="c2"
    />
    <span
      class=""
    >
      one
    </span>
  </li>
  <li
    class="c1"
  >
    <span
      class="c2"
    />
    <span
      class=""
    >
      two
    </span>
  </li>
  <li
    class="c1"
  >
    <span
      class="c2"
    />
    <span
      class=""
    >
      three
    </span>
  </li>
</ul>
`;
